### PR TITLE
[ilm] change remove-policy-from-index http method from DELETE to POST

### PR DESCRIPTION
**Pull request by talevy**
The remove-ilm-from-index API was using the DELETE http method
to signify that something is being removed. Although, metadata
about ILM for the index is being deleted, no entity/resource
is being deleted during this operation. POST is more in line with
what this API is actually doing, it is modifying the metadata for
an index.

cc @bmcconaghy

### CODE REVIEW

**Comment by talevy** [comment]
I believe that an additional URL path change should occur alongside the method change. 

I'd like to use this PR to discuss.

One thing I would like to surface is:

```
POST _ilm
```

The above does not make it clear as to what is happening when called.

Should we add a literal `remove` to the path somewhere to make it clearer?

**Comment by dakrone** [comment]
I definitely don't like it as just `POST /foo/_ilm` to remove the policy. I think at the very minimum we should be explicit and do `POST /foo/_ilm/remove` or `POST /foo/_ilm/policy/remove` (probably the former)

**Comment by talevy** [comment]
@dakrone, I'm in agreement (with the former). aka just appending `/remove` to the path 👍

**Comment by gwbrown** [comment]
++ for making this `<index_name>/_ilm/remove`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ static Request deleteLifecyclePolicy(DeleteLifecyclePolicyRequest deleteLifecycl
     static Request removeIndexLifecyclePolicy(RemoveIndexLifecyclePolicyRequest removePolicyRequest) {
         String[] indices = removePolicyRequest.indices() == null ?
                 Strings.EMPTY_ARRAY : removePolicyRequest.indices().toArray(new String[] {});
-        Request request = new Request(HttpDelete.METHOD_NAME,
+        Request request = new Request(HttpPost.METHOD_NAME,
```
Comment (gwbrown) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java]: You'll also need to change this in [the associated test](https://github.com/elastic/elasticsearch/blob/216c761a5dd81e65ca9826fb0793ddc53cd2ff27/client/rest-high-level/src/test/java/org/elasticsearch/client/IndexLifecycleRequestConvertersTests.java#L102)

Comment (talevy) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/IndexLifecycleRequestConverters.java]: damn. test fail. I ran

```
./gradlew ':client:rest-high-level:test' -Dtests.class=org.elasticsearch.client.indexlifecycle*
```

which doesn't include those! thanks @gwbrown

**Review by colings86 — APPROVED** [review]

**Comment by talevy** [comment]
thanks for providing your feedback @dakrone, @colings86, @gwbrown!

**Pull request merged**